DEVELOPMENT.md

```diff
@@ -32,59 +32,6 @@
    aio app add service
```

Comment:
I cannot leave the comment to the unchanged part, but please remove `(when this is promoted to a github, to be kept in CEXT-3863)` when possible:
https://github.com/adobe/commerce-checkout-starter-kit/blob/c700c66b761a37573efebb7de8a7adef53cf6997/DEVELOPMENT.md#initial-configuration-when-this-is-promoted-to-a-github-to-be-kept-in-cext-3863